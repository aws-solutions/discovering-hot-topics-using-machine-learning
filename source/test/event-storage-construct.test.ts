/**********************************************************************************************************************
 *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.                                           *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                     *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { Template } from 'aws-cdk-lib/assertions';
import { Bucket } from 'aws-cdk-lib/aws-s3';
import { Stack } from 'aws-cdk-lib';
import { EventStorage } from '../lib/storage/event-storage-construct';


test ('Event Storage Construct with props', () => {
    const stack = new Stack();
    new EventStorage(stack, 'EventStorageTest', {
        compressionFormat: 'UNCOMPRESSED',
        prefix: 'test-prefix/',
        s3Bucket: new Bucket(stack, 'testBucket')
    });
    });

test ('Event Storage Construct without props', () => {
    const stack = new Stack();
    const eventStorage = new EventStorage(stack, 'EventStorageTest', {
        compressionFormat: 'GZIP',
        s3Bucket: new Bucket(stack, 'testBucket')
    });
    });

test ('Event Storage Construct with Lambda processor', () => {
    const stack = new Stack();
    const eventStorage = new EventStorage(stack, 'EventStorageTest', {
        compressionFormat: 'GZIP',
        processor: true,
        prefix: 'testPrefix/',
        s3Bucket: new Bucket(stack, 'testBucket')
    });
    });