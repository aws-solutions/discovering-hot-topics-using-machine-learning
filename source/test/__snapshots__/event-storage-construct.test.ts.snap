// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Storage Construct with Lambda processor 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9ArtifactHashF863F156": Object {
      "Description": "Artifact hash for asset \\"8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9\\"",
      "Type": "String",
    },
    "AssetParameters8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9S3BucketFB108141": Object {
      "Description": "S3 bucket for asset \\"8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9\\"",
      "Type": "String",
    },
    "AssetParameters8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9S3VersionKeyB0700792": Object {
      "Description": "S3 key for asset version \\"8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "EventStorageTestFirehoseRole9A5FD08D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "firehose.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "EventStorageTestFirehoseRoleDefaultPolicyE272AE26": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "testBucketDF4D7D1A",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "testBucketDF4D7D1A",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventStorageTestFirehoseRoleDefaultPolicyE272AE26",
        "Roles": Array [
          Object {
            "Ref": "EventStorageTestFirehoseRole9A5FD08D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "EventStorageTestKinesisFirehoseC7A32BA8": Object {
      "DependsOn": Array [
        "EventStorageTestFirehoseRoleDefaultPolicyE272AE26",
        "EventStorageTestFirehoseRole9A5FD08D",
      ],
      "Properties": Object {
        "DeliveryStreamEncryptionConfigurationInput": Object {
          "KeyType": "AWS_OWNED_CMK",
        },
        "DeliveryStreamType": "DirectPut",
        "ExtendedS3DestinationConfiguration": Object {
          "BucketARN": Object {
            "Fn::GetAtt": Array [
              "testBucketDF4D7D1A",
              "Arn",
            ],
          },
          "BufferingHints": Object {
            "IntervalInSeconds": 600,
            "SizeInMBs": 64,
          },
          "CloudWatchLoggingOptions": Object {
            "Enabled": true,
            "LogGroupName": Object {
              "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
            },
            "LogStreamName": Object {
              "Ref": "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1",
            },
          },
          "CompressionFormat": "GZIP",
          "EncryptionConfiguration": Object {
            "KMSEncryptionConfig": Object {
              "AWSKMSKeyARN": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":kms:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":alias/aws/s3",
                  ],
                ],
              },
            },
          },
          "Prefix": "testPrefix/",
          "ProcessingConfiguration": Object {
            "Enabled": true,
            "Processors": Array [
              Object {
                "Parameters": Array [
                  Object {
                    "ParameterName": "LambdaArn",
                    "ParameterValue": Object {
                      "Fn::GetAtt": Array [
                        "EventStorageTestLambdaFunction0DDD544F",
                        "Arn",
                      ],
                    },
                  },
                ],
                "Type": "Lambda",
              },
            ],
          },
          "RoleARN": Object {
            "Fn::GetAtt": Array [
              "EventStorageTestFirehoseRole9A5FD08D",
              "Arn",
            ],
          },
        },
      },
      "Type": "AWS::KinesisFirehose::DeliveryStream",
    },
    "EventStorageTestKinesisFirehoseKinesisFirehosePolicy8524C4A2": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:AbortMultipartUpload",
                "s3:GetBucketLocation",
                "s3:GetObject",
                "s3:ListBucket",
                "s3:ListBucketMultipartUploads",
                "s3:PutObject",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "testBucketDF4D7D1A",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "testBucketDF4D7D1A",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "logs:PutLogEvents",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:",
                    Object {
                      "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
                    },
                    ":log-stream:",
                    Object {
                      "Ref": "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventStorageTestKinesisFirehoseKinesisFirehosePolicy8524C4A2",
        "Roles": Array [
          Object {
            "Ref": "EventStorageTestKinesisFirehoseKinesisFirehoseRoleD2EAD075",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "EventStorageTestKinesisFirehoseKinesisFirehoseRoleD2EAD075": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "firehose.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F": Object {
      "DeletionPolicy": "Retain",
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W86",
              "reason": "Retention period for CloudWatchLogs LogGroups are set to 'Never Expire' to preserve customer data indefinitely",
            },
            Object {
              "id": "W84",
              "reason": "By default CloudWatchLogs LogGroups data is encrypted using the CloudWatch server-side encryption keys (AWS Managed Keys)",
            },
          ],
        },
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": Object {
          "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
        },
      },
      "Type": "AWS::Logs::LogStream",
      "UpdateReplacePolicy": "Retain",
    },
    "EventStorageTestLambdaFunction0DDD544F": Object {
      "DependsOn": Array [
        "EventStorageTestLambdaFunctionServiceRoleDefaultPolicy51A8DD20",
        "EventStorageTestLambdaFunctionServiceRole28BD7F58",
      ],
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W58",
              "reason": "Lambda functions has the required permission to write CloudWatch Logs. It uses custom policy instead of arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole with tighter permissions.",
            },
            Object {
              "id": "W89",
              "reason": "This is not a rule for the general case, just for specific use cases/industries",
            },
            Object {
              "id": "W92",
              "reason": "Impossible for us to define the correct concurrency for clients",
            },
          ],
        },
      },
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9S3BucketFB108141",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9S3VersionKeyB0700792",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters8cca4eb81ccb0a6854e51ac804bb1d3efad1936ca459e803fe3cd3f4b3054ed9S3VersionKeyB0700792",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "EventStorageTestLambdaFunctionServiceRole28BD7F58",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "TracingConfig": Object {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "EventStorageTestLambdaFunctionServiceRole28BD7F58": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "logs:CreateLogGroup",
                    "logs:CreateLogStream",
                    "logs:PutLogEvents",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:",
                        Object {
                          "Ref": "AWS::Partition",
                        },
                        ":logs:",
                        Object {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":log-group:/aws/lambda/*",
                      ],
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "LambdaFunctionServiceRolePolicy",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "EventStorageTestLambdaFunctionServiceRoleDefaultPolicy51A8DD20": Object {
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W12",
              "reason": "Lambda needs the following minimum required permissions to send trace data to X-Ray and access ENIs in a VPC.",
            },
          ],
        },
      },
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventStorageTestLambdaFunctionServiceRoleDefaultPolicy51A8DD20",
        "Roles": Array [
          Object {
            "Ref": "EventStorageTestLambdaFunctionServiceRole28BD7F58",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "EventStorageTestLambdaProcessor4069EDCC": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "lambda:InvokeFunction",
                "lambda:GetFunctionConfiguration",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "EventStorageTestLambdaFunction0DDD544F",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventStorageTestLambdaProcessor4069EDCC",
        "Roles": Array [
          Object {
            "Ref": "EventStorageTestFirehoseRole9A5FD08D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testBucketDF4D7D1A": Object {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;

exports[`Event Storage Construct with props 1`] = `
Object {
  "Resources": Object {
    "EventStorageTestFirehoseRole9A5FD08D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "firehose.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "EventStorageTestFirehoseRoleDefaultPolicyE272AE26": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "testBucketDF4D7D1A",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "testBucketDF4D7D1A",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventStorageTestFirehoseRoleDefaultPolicyE272AE26",
        "Roles": Array [
          Object {
            "Ref": "EventStorageTestFirehoseRole9A5FD08D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "EventStorageTestKinesisFirehoseC7A32BA8": Object {
      "DependsOn": Array [
        "EventStorageTestFirehoseRoleDefaultPolicyE272AE26",
        "EventStorageTestFirehoseRole9A5FD08D",
      ],
      "Properties": Object {
        "DeliveryStreamEncryptionConfigurationInput": Object {
          "KeyType": "AWS_OWNED_CMK",
        },
        "DeliveryStreamType": "DirectPut",
        "ExtendedS3DestinationConfiguration": Object {
          "BucketARN": Object {
            "Fn::GetAtt": Array [
              "testBucketDF4D7D1A",
              "Arn",
            ],
          },
          "BufferingHints": Object {
            "IntervalInSeconds": 600,
            "SizeInMBs": 64,
          },
          "CloudWatchLoggingOptions": Object {
            "Enabled": true,
            "LogGroupName": Object {
              "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
            },
            "LogStreamName": Object {
              "Ref": "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1",
            },
          },
          "CompressionFormat": "UNCOMPRESSED",
          "EncryptionConfiguration": Object {
            "KMSEncryptionConfig": Object {
              "AWSKMSKeyARN": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":kms:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":alias/aws/s3",
                  ],
                ],
              },
            },
          },
          "Prefix": "test-prefix/",
          "RoleARN": Object {
            "Fn::GetAtt": Array [
              "EventStorageTestFirehoseRole9A5FD08D",
              "Arn",
            ],
          },
        },
      },
      "Type": "AWS::KinesisFirehose::DeliveryStream",
    },
    "EventStorageTestKinesisFirehoseKinesisFirehosePolicy8524C4A2": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:AbortMultipartUpload",
                "s3:GetBucketLocation",
                "s3:GetObject",
                "s3:ListBucket",
                "s3:ListBucketMultipartUploads",
                "s3:PutObject",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "testBucketDF4D7D1A",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "testBucketDF4D7D1A",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "logs:PutLogEvents",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:",
                    Object {
                      "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
                    },
                    ":log-stream:",
                    Object {
                      "Ref": "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventStorageTestKinesisFirehoseKinesisFirehosePolicy8524C4A2",
        "Roles": Array [
          Object {
            "Ref": "EventStorageTestKinesisFirehoseKinesisFirehoseRoleD2EAD075",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "EventStorageTestKinesisFirehoseKinesisFirehoseRoleD2EAD075": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "firehose.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F": Object {
      "DeletionPolicy": "Retain",
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W86",
              "reason": "Retention period for CloudWatchLogs LogGroups are set to 'Never Expire' to preserve customer data indefinitely",
            },
            Object {
              "id": "W84",
              "reason": "By default CloudWatchLogs LogGroups data is encrypted using the CloudWatch server-side encryption keys (AWS Managed Keys)",
            },
          ],
        },
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": Object {
          "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
        },
      },
      "Type": "AWS::Logs::LogStream",
      "UpdateReplacePolicy": "Retain",
    },
    "testBucketDF4D7D1A": Object {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;

exports[`Event Storage Construct without props 1`] = `
Object {
  "Resources": Object {
    "EventStorageTestFirehoseRole9A5FD08D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "firehose.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "EventStorageTestFirehoseRoleDefaultPolicyE272AE26": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "testBucketDF4D7D1A",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "testBucketDF4D7D1A",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventStorageTestFirehoseRoleDefaultPolicyE272AE26",
        "Roles": Array [
          Object {
            "Ref": "EventStorageTestFirehoseRole9A5FD08D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "EventStorageTestKinesisFirehoseC7A32BA8": Object {
      "DependsOn": Array [
        "EventStorageTestFirehoseRoleDefaultPolicyE272AE26",
        "EventStorageTestFirehoseRole9A5FD08D",
      ],
      "Properties": Object {
        "DeliveryStreamEncryptionConfigurationInput": Object {
          "KeyType": "AWS_OWNED_CMK",
        },
        "DeliveryStreamType": "DirectPut",
        "ExtendedS3DestinationConfiguration": Object {
          "BucketARN": Object {
            "Fn::GetAtt": Array [
              "testBucketDF4D7D1A",
              "Arn",
            ],
          },
          "BufferingHints": Object {
            "IntervalInSeconds": 600,
            "SizeInMBs": 64,
          },
          "CloudWatchLoggingOptions": Object {
            "Enabled": true,
            "LogGroupName": Object {
              "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
            },
            "LogStreamName": Object {
              "Ref": "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1",
            },
          },
          "CompressionFormat": "GZIP",
          "EncryptionConfiguration": Object {
            "KMSEncryptionConfig": Object {
              "AWSKMSKeyARN": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":kms:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":alias/aws/s3",
                  ],
                ],
              },
            },
          },
          "RoleARN": Object {
            "Fn::GetAtt": Array [
              "EventStorageTestFirehoseRole9A5FD08D",
              "Arn",
            ],
          },
        },
      },
      "Type": "AWS::KinesisFirehose::DeliveryStream",
    },
    "EventStorageTestKinesisFirehoseKinesisFirehosePolicy8524C4A2": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:AbortMultipartUpload",
                "s3:GetBucketLocation",
                "s3:GetObject",
                "s3:ListBucket",
                "s3:ListBucketMultipartUploads",
                "s3:PutObject",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "testBucketDF4D7D1A",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "testBucketDF4D7D1A",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "logs:PutLogEvents",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:",
                    Object {
                      "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
                    },
                    ":log-stream:",
                    Object {
                      "Ref": "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventStorageTestKinesisFirehoseKinesisFirehosePolicy8524C4A2",
        "Roles": Array [
          Object {
            "Ref": "EventStorageTestKinesisFirehoseKinesisFirehoseRoleD2EAD075",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "EventStorageTestKinesisFirehoseKinesisFirehoseRoleD2EAD075": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "firehose.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F": Object {
      "DeletionPolicy": "Retain",
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W86",
              "reason": "Retention period for CloudWatchLogs LogGroups are set to 'Never Expire' to preserve customer data indefinitely",
            },
            Object {
              "id": "W84",
              "reason": "By default CloudWatchLogs LogGroups data is encrypted using the CloudWatch server-side encryption keys (AWS Managed Keys)",
            },
          ],
        },
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "EventStorageTestKinesisFirehosefirehoseloggroupfirehoselogstream1C275BD1": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": Object {
          "Ref": "EventStorageTestKinesisFirehosefirehoseloggroup0CD5B15F",
        },
      },
      "Type": "AWS::Logs::LogStream",
      "UpdateReplacePolicy": "Retain",
    },
    "testBucketDF4D7D1A": Object {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
